File: audiobook/src/main/java/de/ph1b/audiobook/mediaplayer/MediaPlayerInterface.java
package de.ph1b.audiobook.mediaplayer;


import android.content.Context;
import android.media.MediaPlayer;

import java.io.IOException;

public interface MediaPlayerInterface {

    void release();

    void start();

    void reset();

    void prepare() throws IOException;

    void seekTo(int ms);

    int getCurrentPosition();

    void pause();

    float getPlaybackSpeed();

    void setPlaybackSpeed(float speed);

    @SuppressWarnings("RedundantThrows")
    void setDataSource(String source) throws IOException;

    void setOnErrorListener(MediaPlayer.OnErrorListener onErrorListener);

    void setOnCompletionListener(OnCompletionListener onCompletionListener);

    void setWakeMode(Context context, int mode);

    int getDuration();

    interface OnCompletionListener {
        void onCompletion();
    }
}

File: audiobook/src/main/kotlin/de/ph1b/audiobook/mediaplayer/MediaPlayerInterface.kt
package de.ph1b.audiobook.mediaplayer


import android.content.Context
import rx.Observable
import java.io.IOException


/**
 * Basic interface defining our interaction with the media player implementation.
 */
interface MediaPlayerInterface {

    fun release()

    fun start()

    fun reset()

    @Throws(IOException::class)
    fun prepare()

    var currentPosition: Int

    fun pause()

    var playbackSpeed: Float

    @Throws(IOException::class)
    fun setDataSource(source: String)

    val errorObservable: Observable<Unit>

    val completionObservable : Observable<Unit>

    fun setWakeMode(context: Context, mode: Int)

    val duration: Int
}
